master/internal/resourcemanagers/resource_managers.go

```diff
@@ -0,0 +1,76 @@
+package resourcemanagers
+
+import (
+	"crypto/tls"
+	"time"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+const (
+	actionCoolDown = 500 * time.Millisecond
+)
+
+// schedulerTick periodically triggers the scheduler to act.
+type schedulerTick struct{}
+
+// ResourceManagers manages the configured resource providers.
+// Currently support only one resource provider at a time.
+type ResourceManagers struct {
+	ref *actor.Ref
+}
+
+// NewResourceManagers creates an instance of ResourceManagers.
+func NewResourceManagers(
+	system *actor.System,
+	rmConfig *ResourceManagerConfig,
+	poolsConfig *ResourcePoolsConfig,
+	cert *tls.Certificate,
+) *ResourceManagers {
+	var ref *actor.Ref
+	switch {
+	case rmConfig.DeterminedRM != nil:
+		ref, _ = system.ActorOf(
+			actor.Addr("determinedRM"),
+			newDeterminedResourceManager(rmConfig.DeterminedRM, poolsConfig, cert),
+		)
+
+	case rmConfig.KubernetesRM != nil:
+		ref, _ = system.ActorOf(
+			actor.Addr("kubernetesRM"),
+			newKubernetesResourceManager(rmConfig.KubernetesRM),
+		)
+
+	default:
+		panic("no expected resource manager config is defined")
```

Comment:
ditto

Reply:
same: we have already validated the configuration so we shouldn't handle that as errors here.